### PR TITLE
Custom gnmi_cli for device simulator and stratum

### PR DESCRIPTION
**Pull request by adibrastegarnia**

### CODE REVIEW

File: cmd/gnmi_cli/gnmi_cli.go [hunk 1]
```diff
@@ -212,13 +232,36 @@ func executeGet(ctx context.Context) error {
...
-		TLS:         q.TLS,
-	})
+
+	if strings.Compare(*encodingType, "JSON") == 0 {
+		r.Encoding = gpb.Encoding_JSON
+
+		// For the stratum
+	} else if strings.Compare(*encodingType, "PROTO") == 0 {
+		r.Encoding = gpb.Encoding_PROTO
+	}
```
Comment (SeanCondon) [inline cmd/gnmi_cli/gnmi_cli.go:244]: should give an error if it's anything else

File: cmd/gnmi_cli/gnmi_cli.go [hunk 2]
```diff
@@ -64,7 +64,9 @@ var (
 	queryType   = flag.String("query_type", client.Once.String(), "Type of result, one of: (o, once, p, polling, s, streaming).")
 	queryAddr   = flags.NewStringList(&q.Addrs, nil)
 
-	reqProto = flag.String("proto", "", "Text proto for gNMI request.")
+	reqProto     = flag.String("proto", "", "Text proto for gNMI request.")
+	encodingType = flag.String("encodingType", "JSON", "Request Encoding Type")
```
Comment (SeanCondon) [inline cmd/gnmi_cli/gnmi_cli.go]: In description show `JSON|PROTO` as these are the only 2 accepted types

File: cmd/gnmi_cli/gnmi_cli.go [hunk 3]
```diff
@@ -64,7 +64,9 @@ var (
...
+	tlsDisabled  = flag.Bool("tlsDisabled", false, "When set, the TLS will be disabled for gRPC connection")
```
Comment (SeanCondon) [inline cmd/gnmi_cli/gnmi_cli.go]: should say that `caCert, clientCert & clientKey` will be ignored

**Review by SeanCondon — CHANGES_REQUESTED** [review]

File: cmd/gnmi_cli/gnmi_cli.go (hunk 1 above)
Comment (adibrastegarnia) [reply cmd/gnmi_cli/gnmi_cli.go:244]: Done

File: cmd/gnmi_cli/gnmi_cli.go (hunk 2 above)
Comment (adibrastegarnia) [reply cmd/gnmi_cli/gnmi_cli.go]: Done

File: cmd/gnmi_cli/gnmi_cli.go (hunk 3 above)
Comment (adibrastegarnia) [reply cmd/gnmi_cli/gnmi_cli.go]: Done

**Review by SeanCondon — APPROVED** [review]
lgtm

**Pull request merged**